### PR TITLE
Configure `0640` permissions to `vpn` container volumes in `kube-apiserver`

### PR DESCRIPTION
**Pull request by AleksandarSavchev**
**How to categorize this PR?**

/area compliance
/area security
/kind enhancement

**What this PR does / why we need it**:
This PR enhances security by removing `group` write access of mounted files from the `vpn` container volumes. These files are `key` files which we want permissions `0640` or more restrictive.

This PR completes leftover from https://github.com/gardener/gardener/pull/8790

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by AleksandarSavchev** [comment]
Note: `DefaultMode` was set to `400` which are `0620` permissions, however there is a bug/ feature in Kubernetes where when `.spec.securityContext.fsGroup` is set, mounted volume files have group read permission added to them as well, hence the files actually were with `0660` permissions on a working cluster. ref[ issue](https://github.com/kubernetes/kubernetes/issues/57923) [comment](https://github.com/kubernetes/kubernetes/issues/57923#issuecomment-1380630750).

This PR simply removed the write group access to the `vpn-seed-client` volume files.

File: pkg/component/kubernetes/apiserver/deployment.go [hunk 1]
```diff
@@ -697,7 +697,7 @@ func (k *kubeAPIServer) handleVPNSettingsHA(
 			Name: volumeNameVPNSeedClient,
 			VolumeSource: corev1.VolumeSource{
 				Projected: &corev1.ProjectedVolumeSource{
-					DefaultMode: ptr.To[int32](400),
+					DefaultMode: ptr.To[int32](0640),
```
Comment (ScheererJ) [inline pkg/component/kubernetes/apiserver/deployment.go:700]: Thanks a lot for switching to octal representation, which is more obvious for file permissions.

However, I do not get why you add the group read permission. The previous permissions were likely incorrect, but I do not see any reason why there should be group permissions at all. Therefore, I would vote for `0600` or even `0400` instead.

Please note that previously the effective file permission seemed to be `0660`, e.g.

```
root@gardener-local-control-plane:/proc/145868/root/srv/secrets/vpn-client/..2024_07_03_08_28_06.1483950102# ls -lisa
total 12
2 0 drwxr-sr-x 2 root 65532  100 Jul  3 08:28 .
1 0 drwxrwsrwt 3 root 65532  140 Jul  3 08:28 ..
3 4 -rw-rw---- 1 root 65532 1424 Jul  3 08:28 ca.crt
4 4 -rw-rw---- 1 root 65532 1464 Jul  3 08:28 tls.crt
5 4 -rw-rw---- 1 root 65532 2459 Jul  3 08:28 tls.key
```

This was most likely to the `fsGroup` setting in the `securityContext`. Even with a `DefaultMode` of `0400` there was still group read permission.

TLDR: If you just want to unify the access pattern `0640` seems reasonable as it is used all over the place. However, if that is your goal please also adapt line 662 and add the missing `DefaultMode` in line 737. In case you want to improve security, group read should not be required for the VPN components and should be removed from security point of view.

**Review by ScheererJ — CHANGES_REQUESTED** [review]

File: pkg/component/kubernetes/apiserver/deployment.go (hunk 1 above)
Comment (AleksandarSavchev) [reply pkg/component/kubernetes/apiserver/deployment.go:700]: > If you just want to unify the access pattern 0640 seems reasonable as it is used all over the place.

Yes, lets keep the `0640` pattern. Moreover I think it would be a better representation since as mentioned above and in my [comment as well](https://github.com/gardener/gardener/pull/10053#issuecomment-2205409353) the `fsGroup` setting adds the group read access to volume mounts and it would be better if we set it explicitly.

> However, if that is your goal please also adapt line 662 and add the missing DefaultMode in line 737.

I have now set `0640` permissions for the suggested volumes.

**Review by ScheererJ — APPROVED** [review]
/lgtm
/approve
